core/modules/modulesconf.py

```diff
@@ -0,0 +1,46 @@
+from viur.core.bones import StringBone
+from viur.core.tasks import StartupTask
+from viur.core import conf, db
+from viur.core.i18n import translate
+from viur.core.skeleton import Skeleton, SkeletonInstance
+from viur.core.prototypes import List, BasicApplication
+
+
+class ModulesConfSkel(Skeleton):
+    kindName = "viur-modules-conf"
+    name = StringBone(descr=translate("modulename"), readOnly=True)
+    help_text = StringBone(descr=translate("module helptext"))
+    help_text_add = StringBone(descr=translate("add helptext"))
+    help_text_edit = StringBone(descr=translate("edit helptext"))
+    # seo.....
+
+
+class ModulesConf(List):
+    """
+        This module is for viur internal purposes only.
+        It lists all other modules to be able to provide them with help texts.
+    """
+    kindName = "viur-modules-conf"
+    accessRights = ["manage"]
+
+    def canAdd(self):
+        return False
+
+    def canDelete(self, skel: SkeletonInstance) -> bool:
+        return False
+
+
+@StartupTask
+def read_all_modules():
+    for module_name in dir(conf["viur.mainApp"].vi):
```

Comment:
What happens to modules that are being removed? The config then stays... if the config is used for general module retrieval, it should be removed.

Reply:
Deleting entities is a very bad idea here. If you add a new module and deploy it as version X, you can already prepare the data there. But in standard/live version Y the module is still unknown, so it would be deleted. X would create it again at the next instance start and Y delete it again ….
The same happens when you do a rollback after deleting a module or work with different states in a development system.

Reply:
Maybe we can take out the delete part for now. In my opinion it is not so bad if the record remains in the database.